packages/yourloops/lib/patient/hook.tsx

```diff
@@ -98,20 +98,28 @@ export default function usePatientProviderCustomHook(): PatientContextResult {
   const getPatientById = useCallback(userId => patients.find(patient => patient.userid === userId), [patients])
 
   const filterPatients = useCallback((filterType: PatientFilterTypes, search: string, flaggedPatients: string[]) => {
-    let res = PatientUtils.extractPatients(patients, filterType, flaggedPatients)
-    const searchByName = search.length > 0
-    if (searchByName) {
+    const filteredPatients = PatientUtils.extractPatients(patients, filterType, flaggedPatients)
+    if (search.length > 0) {
       const searchText = search.toLocaleLowerCase()
-      res = res.filter((patient: Patient): boolean => {
+      const birthdateAsString = searchText.slice(0, 10)
+      if (moment(birthdateAsString, 'DD/MM/YYYY').toDate().getTime()) {
+        const firstNameOrLastName = searchText.slice(10).trimStart()
+        return filteredPatients.filter((patient: Patient): boolean => {
```

Comment:
Putting this `filter()` in a dedicated method would allow people to better understand its intent (overall, I feel like there is a lot of code to read in this `if()` condition, and splitting it into smaller methods would improve the readability)